packages/db/src/translate/granules.ts

```diff
@@ -0,0 +1,76 @@
+import Knex from 'knex';
+
+import { getRecordCumulusId } from '../database';
+import { PdrPgModel } from '../models/pdr';
+import { PostgresGranule } from '../types/granule';
+import { PostgresExecutionRecord } from '../types/execution';
+import { ProviderPgModel } from '../models/provider';
+import { tableNames } from '../tables';
+
+/**
+ * Generate a Postgres rule record from a DynamoDB record.
+ *
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} dynamoRecord
+ *   Record from DynamoDB
+ * @param {Object} knex - Knex client for reading from RDS database
+ * @param {number} collectionCumulusId - Collection Cumulus Id
+ * @param {Object} providerPgModel - Instance of the provider database model
+ * @param {Object} pdrPgModel - Instance of the pdr database model
+ * @returns {Object} A granule PG record
+ */
+export const translateApiGranuleToPostgresGranule = async (
+  dynamoRecord: AWS.DynamoDB.DocumentClient.AttributeMap,
+  knex: Knex,
+  collectionCumulusId: number,
+  providerPgModel = new ProviderPgModel(),
+  pdrPgModel = new PdrPgModel()
+): Promise<PostgresGranule> => {
+  const granuleRecord: PostgresGranule = {
+    granule_id: dynamoRecord.granuleId,
+    status: dynamoRecord.status,
+    collection_cumulus_id: collectionCumulusId,
+    published: dynamoRecord.published,
+    duration: dynamoRecord.duration,
+    time_to_archive: dynamoRecord.timeToArchive,
+    time_to_process: dynamoRecord.timeToPreprocess,
+    product_volume: dynamoRecord.productVolume,
+    error: dynamoRecord.error,
+    cmr_link: dynamoRecord.cmrLink,
+    execution_cumulus_id: dynamoRecord.execution
+      ? await getRecordCumulusId<PostgresExecutionRecord>(
```

Comment:
Should be using the execution PG model `getRecordCumulusId` method?

Reply:
[`c58cf1b` (#2051)](https://github.com/nasa/cumulus/pull/2051/commits/c58cf1bbdd841e01b1621ee0b002c6a4c0b8d210)